File: app/src/main/java/com/ducnd/demorealmmvp/remote/interact/main/BasePresenterM.java
package com.ducnd.demorealmmvp.remote.interact.main;

import com.ducnd.demorealmmvp.remote.interact.interf.IAccountInteraction;
import com.ducnd.realmmvp.remote.interact.main.BasePresenter;
import com.ducnd.realmmvp.ui.base.BaseViewUI;

import org.jetbrains.annotations.NotNull;

/**
 * Created by ducnd on 8/11/17.
 */

public abstract class BasePresenterM<V extends BaseViewUI> extends BasePresenter<V> {
    protected IAccountInteraction mAccountInteraction;

    public BasePresenterM(@NotNull V view, @NotNull IAccountInteraction accountInteraction) {
        super(view);
        mAccountInteraction = accountInteraction;
    }
}

File: app/src/main/java/com/ducnd/demorealmmvp/remote/interact/main/BasePresenterM.kt
package com.ducnd.demorealmmvp.remote.interact.main

import android.graphics.Bitmap
import com.ducnd.demorealmmvp.remote.interact.interf.IAccountInteraction
import com.ducnd.demorealmmvp.remote.interact.interf.IBasePresenterM
import com.ducnd.demorealmmvp.remote.model.MediaInfo
import com.ducnd.realmmvp.remote.interact.main.BasePresenter
import com.ducnd.realmmvp.ui.base.BaseViewUI
import io.reactivex.disposables.Disposable

/**
 * Created by ducnd on 8/11/17.
 */

abstract class BasePresenterM<V : BaseViewUI>(view: V, protected var mAccountInteraction: IAccountInteraction) : BasePresenter<V>(view), IBasePresenterM {
    override fun saveMediaInfoAtFreeThread(bitmap: Bitmap, localFolderMedia: String, linkImage: String): Disposable {
        return mAccountInteraction.saveMediaInfoAtFreeThread(bitmap, localFolderMedia, linkImage)
    }

    override fun findImageMediaInfoAtMainThread(fieldName: String, valueCondition: String): MediaInfo? {
        return mAccountInteraction.findItemAtMainThread(MediaInfo::class.java, fieldName, valueCondition)
    }
}
